### PR TITLE
Add support for UForge API Keys

### PR DESCRIPTION
**Pull request by segalaj**
:warning: need https://github.com/usharesoft/ussclicore/pull/25

### CODE REVIEW

File: tests/unit/utils/test_credentials.py [hunk 1]
```diff
@@ -1,56 +1,152 @@
...
+
+    def test_is_api_auth_valid_return_false_when_no_public_key(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password=None, publicKey=None, secretKey="my_secret")
+
+        # when
+        valid = cred.is_api_auth_valid()
+
+        # then
+        self.assertFalse(valid)
```
Comment (lqueiroga) [inline tests/unit/utils/test_credentials.py]: There is not the case return_false_when_no_secret_key

Comment (segalaj) [reply tests/unit/utils/test_credentials.py]: done

File: hammr/utils/credentials.py [hunk 2]
```diff
@@ -56,17 +74,27 @@ def sslAutosigned(self):
...
+            return None
 
     def validate(self):
-        if self.username is None:
-            raise CredentialsException("No user provided.")
-        if self.password is None:
-            raise CredentialsException("No password provided.")
-        if self.url is None:
-            raise CredentialsException("No URL provided.")
+        if self.is_basic_auth_valid() or self.is_api_auth_valid():
```
[the remark this answers is not in the document]
Comment (segalaj) [reply hammr/utils/credentials.py]: done

**Comment by lqueiroga** [comment]
Thanks for the PR !

**Pull request merged**